bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/RuleRegistry.java

```diff
@@ -44,10 +49,19 @@
  * </ul>
  *
  * @author Yordan Mihaylov - Initial Contribution
+ * @author Victor Toni - added search capability via {@link Predicate}
  */
 public interface RuleRegistry extends Registry<Rule, String> {
 
     /**
+     * This method is used to get collection of {@link Rule}s which have the same {@link Predicate}.
+     *
+     * @param predicate to check the rules against
+     * @return collection of {@link Rule}s having matching predicate.
+     */
+    public Collection<Rule> get(Predicate<Rule> predicate);
```

Comment:
I don't think so, the get() method accepts uid as parameter and returns corresponding element (there is a 1:1 relation between parameter and the element). On the other hand the  getAll return collection of elements, all the registries in ESH implement this concept. 
Tags is something that group elements and for me it is normal to return the collection.